linera-service/src/cli_wrappers/util.rs

```diff
@@ -19,3 +19,22 @@ pub async fn get_github_root() -> Result<PathBuf> {
             .to_string(),
     ))
 }
+
+pub async fn get_git_head_commit() -> Result<String> {
+    let mut git_head_commit = Command::new("git")
+        .arg("rev-parse")
+        .arg("HEAD")
+        .spawn_and_wait_for_stdout()
+        .await?;
+    let uncommitted_changes = Command::new("git")
+        .arg("diff-index")
+        .args(["--quiet", "HEAD"])
+        .status()
+        .await?;
+
+    if !uncommitted_changes.success() {
+        git_head_commit = format!("{}-dirty", git_head_commit);
+    }
+
+    Ok(git_head_commit)
+}
```

Comment:
I suggest re-using `linera_base::VersionInfo` here.

Reply:
One caveat though: that could be different if the repo changed after compilation, and the binary is running on the changed repo :thinking:

Reply:
`VersionInfo::default()` would be — but I think it might make sense to add `VersionInfo::get()` that generates the info when called.

Reply:
Oh! That's nice! :tada:

Reply:
I made a PR with this functionality :) https://github.com/linera-io/linera-protocol/pull/1591

Reply:
Sounds good, will wait for the PR to land and will adjust this